code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,36 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("- no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		log_and_message_admins("has generated a DeepmaintRooms map.")
+		break
```

Comment:
Seems a little odd to break on the first iteration without any kind of checks. Could you not just access `wfc_deepmaint_zlevels_by_instance` by the first index instead?
```suggestion
	var/area/map_template/deepmaint_wfc/deepmaint_area = GLOB.wfc_deepmaint_zlevels_by_instance[1]
	deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
	log_and_message_admins("has generated a DeepmaintRooms map.")
```

Reply:
No, because _you're not guaranteed to have an index 1_. The current code handles a failure gracefully and doesn't report success on a failure.

Reply:
I actually thought about this after I turned the PC off. Fair point!